### ISSUE TITLE
crashes when used on ESP32C3

### ISSUE THREAD

**Issue by kaxori** [toitware/toit-pixel-strip#19]
******************************************************************************
Decoding by `jag`, device has version <2.0.0-alpha.90>
******************************************************************************
EXCEPTION error. 
Connection closed
  0: tcp_error_                <sdk>/net/modules/tcp.toit:208:3
  1: TcpSocket_.ensure_state_  <sdk>/net/modules/tcp.toit:75:14
  2: TcpSocket.read            <sdk>/net/modules/tcp.toit:156:16
  3: BufferedReader.more_      <sdk>/reader.toit:82:22
  4: BufferedReader.ensure_    <sdk>/reader.toit:69:14
  5: BufferedReader.can_ensure <sdk>/reader.toit:109:5
  6: Connection.read_request   <pkg:pkg-http>/connection.toit:138:20
  7: Server.run_connection_.<block> <pkg:pkg-http>/server.toit:146:30
  8: Task_.with_deadline_.<block> <sdk>/core/task.toit:203:16
  9: Task_.with_deadline_      <sdk>/core/task.toit:197:3
 10: with_timeout              <sdk>/core/utils.toit:181:24
 11: with_timeout              <sdk>/core/utils.toit:164:12
 12: Server.run_connection_    <pkg:pkg-http>/server.toit:145:7
 13: Server.listen.<block>.<lambda>.<block>.<block> <pkg:pkg-http>/server.toit:113:26
 14: catch.<block>             <sdk>/core/exceptions.toit:124:10
 15: catch                     <sdk>/core/exceptions.toit:122:1
 16: catch                     <sdk>/core/exceptions.toit:97:10
 17: Server.listen.<block>.<lambda>.<block> <pkg:pkg-http>/server.toit:112:18
 18: Server.listen.<block>.<lambda> <pkg:pkg-http>/server.toit:109:38
 19: Server.listen.<block>     <pkg:pkg-http>/server.toit:129:37
 20: Server.listen             <pkg:pkg-http>/server.toit:85:3
 21: serve_incoming_requests   /home/runner/work/jaguar/jaguar/src/jaguar.toit:401:10
 22: run.<block>.<lambda>      /home/runner/work/jaguar/jaguar/src/jaguar.toit:199:10
 23: Task.group.<lambda>.<block> <sdk>/core/task.toit:113:45
 24: Task.group.<lambda>       <sdk>/core/task.toit:110:26
******************************************************************************

**Comment by kaxori** [comment]
RESTART

TEST
Guru Meditation Error: Core  0 panic'ed (Instruction access fault). Exception was unhandled.

Core  0 register dump:
MEPC    : 0x0000000e  RA      : 0x403801ee  SP      : 0x3fc99ac0  GP      : 0x3fc96c00
TP      : 0x3fc8f880  T0      : 0x7fffffff  T1      : 0x0000000f  T2      : 0x00000005
S0/FP   : 0x3fcb9450  S1      : 0x80000009  A0      : 0x5b5b5b5b  A1      : 0x3fc99b38
A2      : 0x5b5b5b5b  A3      : 0x00000004  A4      : 0x600c2000  A5      : 0x00000048
A6      : 0x5b5b5b5b  A7      : 0x5b5b5b5b  S2      : 0x00001881  S3      : 0x00000004
S4      : 0xfffffcc4  S5      : 0x3fcb9018  S6      : 0x3fcb9018  S7      : 0x3fc992b0
S8      : 0x3fca8f30  S9      : 0x00000018  S10     : 0x3fcbc3d4  S11     : 0x3c149000
T3      : 0x3fc9ce30  T4      : 0x0000000f  T5      : 0x00000010  T6      : 0x5b5b5b5b
MSTATUS : 0x00001881  MTVEC   : 0x40380001  MCAUSE  : 0x00000001  MTVAL   : 0x0000000e
MHARTID : 0x00000000

Stack memory:
3fc99ac0: 0x40390100 0x40390116 0x3fc99b40 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99ae0: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99b00: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99b20: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x0000000e 0x0000000f 0x5b5b5b5b
3fc99b40: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99b60: 0x40390100 0x40390116 0x3fc99be0 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99b80: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99ba0: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99bc0: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x0000000d 0x0000000e 0x5b5b5b5b
3fc99be0: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99c00: 0x40390100 0x40390116 0x3fc99c80 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99c20: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99c40: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99c60: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x0000000c 0x0000000d 0x5b5b5b5b
3fc99c80: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99ca0: 0x40390100 0x40390116 0x3fc99d20 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99cc0: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99ce0: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99d00: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x0000000b 0x0000000c 0x5b5b5b5b
3fc99d20: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99d40: 0x40390100 0x40390116 0x3fc99dc0 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99d60: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99d80: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99da0: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x0000000a 0x0000000b 0x5b5b5b5b
3fc99dc0: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99de0: 0x40390100 0x40390116 0x3fc99e60 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99e00: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881
3fc99e20: 0x5b5b5b5b 0x5b5b5b5b 0x00001881 0x00000004 0xfffffcc4 0x3fcb9018 0x3fcb9018 0x3fc992b0
3fc99e40: 0x3fca8f30 0x00000018 0x3fcbc3d4 0x3c149000 0x3fc9ce30 0x00000009 0x0000000a 0x5b5b5b5b
3fc99e60: 0x00000000 0x00000000 0x00000000 0x403889ce 0x00000000 0x00000000 0x3fcb9450 0x403801ee
3fc99e80: 0x40390100 0x40390116 0x3fc99f00 0x3fc96c00 0x3fc8f880 0x7fffffff 0x403889c2 0x00000005
3fc99ea0: 0x3fcb9450 0x80000009 0x00000005 0x3fc99b38 0x5b5b5b5b 0x00000004 0x600c2000 0x00001881



ELF file SHA256: 825780c580628c8e

Rebooting...
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x403870b6
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd5810,len:0x48
load:0x403cc710,len:0x650
load:0x403ce710,len:0x23f0
entry 0x403cc710
[toit] INFO: starting <v2.0.0-alpha.90>
[wifi] DEBUG: connecting
E (1032) wifi:Association refused temporarily, comeback time 1024 mSec
[wifi] WARN: connect failed {reason: unknown reason (203)}
[wifi] DEBUG: closing
[jaguar] WARN: running Jaguar failed due to 'CONNECT_FAILED: unknown reason (203)' (1/3)
[wifi] DEBUG: connecting
[wifi] DEBUG: connected
[wifi] INFO: network address dynamically assigned through dhcp {ip: 192.168.178.77}
[wifi] INFO: dns server address dynamically assigned through dhcp {ip: [192.168.178.1]}
[jaguar] INFO: running Jaguar device 'RGB' (id: 'fa4521cb-b229-411e-98d0-b45250361d86') on 'http://192.168.178.77:9000'

**Comment by kaxori** [comment]
crashes with :
`  neopixels.output r g b`

! same code run on ESP32 !

**Comment by erikcorry** [comment]
Do you think you might be seeing a brown-out. The pixel strings can use a lot of power.

Try running with the pixel strip disconnected.  Since it's a one way protocol, the ESP32C3 will not notice.

**Comment by kaxori** [comment]
No, definitely not.
there is no load connected

**Comment by erikcorry** [comment]
OK, can you share more details on how you are reproducing the failure?
Jaguar version.
C3 config (does it have PSRAM?)
Length of strip and bytes per pixel?
Perhaps a simple program that reproduces the crash?
Is the pin being used for something else (flipping it with GPIO functions doesn't cause an issue?)

**Comment by kaxori** [comment]
jag version
Version:         v1.10.0
SDK version:     v2.0.0-alpha.90
Build date:      2023-07-06T15:43:49Z

**XIAO_ESP32C3**:  https://wiki.seeedstudio.com/XIAO_ESP32C3_Getting_Started/

several GPIO Pins tested: 2, 10 and no strip connected

https://github.com/kaxori/Prj_5_Philosophers/blob/main/Philo.toit

**Comment by erikcorry** [comment]
I could reproduce this on Jaguar v1.10.0 (Toit alpha 90), but it runs correctly on the latest SDKs.
I also reproduced this on Jaguar v1.10.3 (Toit alpha 95)
I also reproduced this on Jaguar v1.10.4 (Toit alpha 95)
On Jaguar v1.11.0 (Toit alpha 97) I saw a hang instead of a crash.  This is the release (alpha 97) that goes from esp-idf 4.4 to esp-idf 5.0.
That issue has now been fixed in Toit v2.0.0-alpha 98, which is not yet released by Jaguar.
